### PR TITLE
Add a NuGet.config pointing to AppVeyor and MyGet feeds

### PR DESCRIPTION
**Pull request by rprouse**
See #1565

### CODE REVIEW

**Comment by rprouse** [comment]
NuGet can't find Cake, I might have made a mistake, but to make sure it is not a temporary failure, I have restarted on of the builds.

File: NuGet.config [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <clear/>
+    <add key="NUnit AppVeyor CI" value="https://ci.appveyor.com/nuget/nunit" />
+    <add key="NUNit MyGet Feed" value="https://www.myget.org/F/nunit/api/v2" />
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
```
Comment (jasonwilliams200OK) [inline NuGet.config]: Can you try dropping `protocolVersion="3"` part?

Comment (jasonwilliams200OK) [inline NuGet.config]: nit: also add the missing end-of-file line feed :wink:

**Comment by rprouse** [comment]
If my latest checkin doesn't fix it, it could be related to NuGet/Home#1283, possibly because of the version of NuGet.exe in our repo or on AppVeyor?

**Comment by jasonwilliams200OK** [comment]
Is `protocolVersion="3"` usage really required? CoreCLR guys don't use it for example: https://github.com/dotnet/coreclr/blob/899d427/tests/src/NuGet.Config.

**Comment by rprouse** [comment]
@jasonwilliams200OK I am not sure about `protocolVersion="3"` I just see it everywhere :smile: 

The nuget restore is now working, but we are now getting a real and unrelated failure on Travis mono lates;

```Errors and Failures
1) Failed : NUnit.Engine.Services.Tests.RuntimeFrameworkServiceTests.SelectRuntimeFramework("mock-assembly.exe","2.0.50727",False)
  String lengths are both 9. Strings differ at index 0.
  Expected: "2.0.50727"
  But was:  "4.0.30319"
  -----------^
at NUnit.Engine.Services.Tests.RuntimeFrameworkServiceTests.SelectRuntimeFramework (System.String assemblyName, System.String expectedVersion, Boolean runAsX86) <0x419e6230 + 0x001ef> in <filename unknown>:0
2) Failed : NUnit.Engine.Services.Tests.RuntimeFrameworkServiceTests.SelectRuntimeFramework("net-2.0/mock-assembly.exe","2.0.50727",False)
  String lengths are both 9. Strings differ at index 0.
  Expected: "2.0.50727"
  But was:  "4.0.30319"
  -----------^
at NUnit.Engine.Services.Tests.RuntimeFrameworkServiceTests.SelectRuntimeFramework (System.String assemblyName, System.String expectedVersion, Boolean runAsX86) <0x419e6230 + 0x001ef> in <filename unknown>:0
3) Failed : NUnit.Engine.Services.Tests.RuntimeFrameworkServiceTests.SelectRuntimeFramework("nunit-agent.exe","2.0.50727",False)
  String lengths are both 9. Strings differ at index 0.
  Expected: "2.0.50727"
  But was:  "4.0.30319"
  -----------^
at NUnit.Engine.Services.Tests.RuntimeFrameworkServiceTests.SelectRuntimeFramework (System.String assemblyName, System.String expectedVersion, Boolean runAsX86) <0x419e6230 + 0x001ef> in <filename unknown>:0
4) Failed : NUnit.Engine.Services.Tests.RuntimeFrameworkServiceTests.SelectRuntimeFramework("nunit-agent-x86.exe","2.0.50727",True)
  String lengths are both 9. Strings differ at index 0.
  Expected: "2.0.50727"
  But was:  "4.0.30319"
  -----------^
at NUnit.Engine.Services.Tests.RuntimeFrameworkServiceTests.SelectRuntimeFramework (System.String assemblyName, System.String expectedVersion, Boolean runAsX86) <0x419e6230 + 0x001ef> in <filename unknown>:0
```

**Comment by rprouse** [comment]
The one AppVeyor failure is because of my branch name.

**Comment by rprouse** [comment]
Can someone look a the Travis latest failure? Has mono been updated breaking our setting the framework to 2.0 for test runs?

**Comment by CharliePoole** [comment]
I'm assuming that the fix @ChrisMaddock did will take care of this. Will you pull it into this branch?

**Comment by rprouse** [comment]
Merged master, waiting for CI

**Pull request merged**